custom_components/mqtt_vacuum_camera/utils/vacuum/mqtt_vacuum_services.py

```
@@ -0,0 +1,515 @@
+            raise ServiceValidationError(f"Error sending command to vacuum: {e}")
+        coordinator.hass.bus.async_fire(
+            f"event_{DOMAIN}.vacuum_clean_zone",
+            {
+                "topic": service_data["topic"],
+                "zones": zone_lists,
+                "repeats": repeats,
+            },
+            context=call.context,
+        )
+    except KeyError as e:
+        raise ServiceValidationError(f"Missing required parameter: {e}")
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)